x/evm/statedb/statedb_test.go

```diff
@@ -0,0 +1,360 @@
+
+				db = statedb.New(db.Context(), db.Keeper(), testTxConfig)
+				require.Equal(t, uint64(1), db.GetNonce(addr2))
+			},
+		},
+		{
+			"success,logs",
+			func(t *testing.T, db *statedb.StateDB) {
+				data := []byte("hello world")
+				db.AddLog(&ethtypes.Log{
+					Address:     addr2,
+					Topics:      []common.Hash{},
+					Data:        data,
+					BlockNumber: 1,
+				})
+				require.Equal(t, 1, len(db.Logs()))
+				expecedLog := &ethtypes.Log{
+					Address:     addr2,
+					Topics:      []common.Hash{},
+					Data:        data,
+					BlockNumber: 1,
+					BlockHash:   common.BigToHash(big.NewInt(10)),
+					TxHash:      common.BigToHash(big.NewInt(11)),
+					TxIndex:     1,
+					Index:       1,
+				}
+				require.Equal(t, expecedLog, db.Logs()[0])
+
+				rev := db.Snapshot()
+
+				db.AddLog(&ethtypes.Log{
+					Address:     addr2,
+					Topics:      []common.Hash{},
+					Data:        data,
+					BlockNumber: 1,
+				})
+				require.Equal(t, 2, len(db.Logs()))
+				require.Equal(t, uint(2), db.Logs()[1].Index)
+
+				db.RevertToSnapshot(rev)
+				require.Equal(t, 1, len(db.Logs()))
+			},
+		},
+		{
+			"success,refund",
+			func(t *testing.T, db *statedb.StateDB) {
+				db.AddRefund(uint64(10))
+				require.Equal(t, uint64(10), db.GetRefund())
+
+				rev := db.Snapshot()
+
+				db.SubRefund(uint64(5))
+				require.Equal(t, uint64(5), db.GetRefund())
+
+				db.RevertToSnapshot(rev)
+				require.Equal(t, uint64(10), db.GetRefund())
+			},
+		},
+		{
+			"success,empty",
+			func(t *testing.T, db *statedb.StateDB) {
+				require.False(t, db.Exist(addr2))
+				require.True(t, db.Empty(addr2))
+
+				db.AddBalance(addr2, big.NewInt(1))
+				require.True(t, db.Exist(addr2))
+				require.False(t, db.Empty(addr2))
+
+				db.SubBalance(addr2, big.NewInt(1))
+				require.True(t, db.Exist(addr2))
+				require.True(t, db.Empty(addr2))
+			},
+		},
+		{
+			"success,suicide commit",
+			func(t *testing.T, db *statedb.StateDB) {
+				code := []byte("hello world")
+				db.SetCode(addr2, code)
+				db.AddBalance(addr2, big.NewInt(1))
+
+				require.True(t, db.Exist(addr2))
+				require.False(t, db.Empty(addr2))
+
+				db.Suicide(addr2)
+				require.True(t, db.HasSuicided(addr2))
+				require.True(t, db.Exist(addr2))
+				require.Equal(t, new(big.Int), db.GetBalance(addr2))
+
+				db.Commit()
+				db = statedb.New(db.Context(), db.Keeper(), testTxConfig)
+				require.True(t, db.Empty(addr2))
+			},
+		},
+		{
+			"success,suicide revert",
+			func(t *testing.T, db *statedb.StateDB) {
+				code := []byte("hello world")
+				db.SetCode(addr2, code)
+				db.AddBalance(addr2, big.NewInt(1))
+
+				rev := db.Snapshot()
+
+				db.Suicide(addr2)
+				require.True(t, db.HasSuicided(addr2))
+
+				db.RevertToSnapshot(rev)
+
+				require.False(t, db.HasSuicided(addr2))
+				require.Equal(t, code, db.GetCode(addr2))
+				require.Equal(t, big.NewInt(1), db.GetBalance(addr2))
+			},
+		},
+		// TODO access list, ForEachStorage
```

Comment:
address this TODO or create issue for follow up

Reply:
linked the https://github.com/tharsis/ethermint/issues/876 here.